control_fun.h

```diff
@@ -228,10 +228,12 @@ class PI2Controller {
     float step(float desired, float measured);
     void set_param(const PI2Param &pi_param);
     void initialize() { ki_sum_ = 0; }
+    bool is_voltage_saturated() const { return std::abs(ki_sum_ == ki_limit_) >= ki_limit_ || 
```

Comment:
I don't quite follow the first logical expression here. It seems to check if the cumulative integral is equal to the limit, which I get, but then if that logical result is >= the limit. Can you help me understand this part?